### ISSUE TITLE
Provide a ML data source for the getConfig() method.

### ISSUE THREAD

**Issue by cmacfarl** [hashicorp/terraform-provider-google#9714]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment. If the issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If the issue is assigned to a user, that user is claiming responsibility for the issue. If the issue is assigned to "hashibot", a community member has claimed the issue already.

### Description

Add the ability to set IAM roles on the tpuServiceAccount by providing a data source for same.

The following API endpoint returns a google managed service account for tpu instances.

```
https://ml.googleapis.com/v1/projects/<your-project-id>:getConfig
```

Sample json return

```
{
  "serviceAccount": "service-<elided>@cloud-ml.google.com.iam.gserviceaccount.com",
  "serviceAccountProject": "<elided>",
  "config": {
    "tpuServiceAccount": "service-<elided>@cloud-tpu.iam.gserviceaccount.com"
  }
}
```

There is currently no way to set IAM roles for the tpuServiceAccount because there is no way to get a handle to the account.

### New or Affected Resource(s)

```
data "google_ml_config" "cfg" {
}
```
Populates attributes: serviceAccount, serviceAccountProject, tpuServiceAccount.

### Potential Terraform Configuration
```
resource "google_project_iam_binding" "tpu_role" {
  project = "<your-project-id>"
  role    = "roles/ml.serviceAgent"
  members = [
    "serviceAccount:${google_ml_config.cfg.tpuServiceAccount}"
  ]
```

### References

Full documentation for the API endpoint [here](https://cloud.google.com/ai-platform/prediction/docs/reference/rest/v1/projects/getConfig)

**Comment by slevenick** [comment]
Hey @cmacfarl I'm not super familiar with this API, but we have a resource that serves a similar function in [google_project_service_identity](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/project_service_identity).

Does that work for your use case? It lets you set up a service account for a GCP service via a single resource.

**Comment by cmacfarl** [comment]
Thanks for the response @slevenick.  

It's not clear that google_project_service_identity serves my use case.  Reading of the documentation seems to indicate that it _generates_ a service account, and what I actually want is the google managed TPU service account attached to the cloud-ml service account as in the json output above.

At any rate, I went ahead and cranked out a trivial provider.

https://registry.terraform.io/providers/cmacfarl/google-ml/latest

It implements getConfig() and simply sets as attributes the three properties identified in the json returned from the endpoint.

**Comment by slevenick** [comment]
Generally the `google_project_service_identity` resource generates *or* gets the service account for a particular service. I would be curious if it returns the same result as the `getConfig` method you linked, but I don't know if it will

**Comment by cmacfarl** [comment]
Yeah, I dropped that resource in a tf file and ran terraform apply against ml.googleapis.com and since it's a resource the plan said it would have the value of the email attribute after it applied the change.  But I didn't want to change the state of my project, so I cancelled the apply.  I suspect however that it would just return the outer cloud-ml email address and not the cloud-tpu email address.   Given that there's only one attribute returned for the resource it's hard to imagine why that attribute would be the cloud-tpu address instead of the cloud-ml address.
